packages/vite/src/node/plugins/dynamicImportVars.ts

```diff
@@ -105,10 +107,20 @@ export async function transformDynamicImport(
   const params = globParams
     ? `, ${JSON.stringify({ ...globParams, import: '*' })}`
     : ''
+
+  let newRawPattern = posix.relative(
+    posix.dirname(importer),
+    await toAbsoluteGlob(rawPattern, root, importer, resolve)
+  )
+
+  if (!/^\.{1,2}\//.test(newRawPattern)) {
+    newRawPattern = `./${newRawPattern}`
+  }
```

Comment:
I don't understand this logic, do we have enough tests for all cases here?
If importer is `/root/dir/nested/importer.js` and `toAbsoluteGlob` returns `/root/foo/bar/{pattern}.js`, then `newRawPattern` ends up being `./../foo/bar/{pattern}.js` 🤔

Reply:
Ah, no.. I missed the `!`. Ok, now I understand 😄